### PR TITLE
[Error handling] Improve logging when fetching Order details from Storage

### PR DESCRIPTION
**Pull request by iamgabrielma**
Part of #8458

## Description
We are not currently logging any errors from operations related to fetching from Storage within the `OrderDetailsResultsControllers` class, which holds the different results controllers used to render the Order Details view. There's some instances of the same in other View-related Order classes, like `OrderLoaderViewController`, `OrdersRootViewController`, or `OrderSearchUICommand`. 

If any of the fetch result operations happens to return `nil`, this fails silently as we're not catching or logging the error, going unnoticed.

**Question:** We also perform a call to [refetchAllResultsControllers](https://github.com/woocommerce/woocommerce-ios/blob/4e0c223fc32db61b33c6e7b1c5c28973cd025f2f/WooCommerce/Classes/ViewModels/Order%20Details/OrderDetailsResultsControllers.swift#L315-L326) in order to re-fetch all the results controllers if necessary, and this involves more than one results controller. I left the optional `try?` in this case as if we wrap that on a `do-catch` block, the first operation that returns nil would also halt the rest, but happy to get any feedback on a better way to deal with those.

## Changes

This PR wraps the different optional `try?` calls to fetch results controller into a `do-catch` block, with a specific `DDLog` message to log the error if needed.

## Testing instructions
- Confirm that Unit Tests pass.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift [hunk 1]
```diff
@@ -261,7 +266,11 @@ private extension OrdersRootViewController {
             self.resetFiltersIfAnyStatusFilterIsNoMoreExisting(orderStatuses: self.statusResultsController.fetchedObjects)
         }
 
-        try? statusResultsController.performFetch()
+        do {
+            try statusResultsController.performFetch()
+        } catch {
+            DDLogError("⛔️ Unable to fetch stored statuse for Site \(siteID): \(error)")
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift]: super nit:
```suggestion
            DDLogError("⛔️ Unable to fetch stored order statuses for Site \(siteID): \(error)")
```

File: WooCommerce/Classes/ViewRelated/Search/Order/OrderSearchUICommand.swift [hunk 2]
```diff
@@ -101,7 +101,11 @@ final class OrderSearchUICommand: SearchUICommand {
 
 private extension OrderSearchUICommand {
     func configureResultsController() {
-        try? statusResultsController.performFetch()
+        do {
+            try statusResultsController.performFetch()
+        } catch {
+            DDLogError("⛔️ Unable to fetch Order search results for Site \(siteID): \(error)")
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Search/Order/OrderSearchUICommand.swift]: nit: I think this results controller is for order statuses?

```suggestion
            DDLogError("⛔️ Unable to fetch Order statuses in order search for Site \(siteID): \(error)")
```

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsResultsControllers.swift [hunk 3]
```diff
@@ -267,7 +287,11 @@ private extension OrderDetailsResultsControllers {
             onReload()
         }
 
-        try? addOnGroupResultsController.performFetch()
+        do {
+            try addOnGroupResultsController.performFetch()
+        } catch {
+            DDLogError("⛔️ Unable to fetch AddOnGroup Site \(siteID): \(error)")
```
Comment (jaclync) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsResultsControllers.swift]: super nit:

```suggestion
            DDLogError("⛔️ Unable to fetch AddOnGroups for Site \(siteID): \(error)")
```

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsResultsControllers.swift [hunk 4]
```diff
@@ -253,7 +269,11 @@ private extension OrderDetailsResultsControllers {
             onReload()
         }
 
-        try? shippingLabelResultsController.performFetch()
+        do {
+            try shippingLabelResultsController.performFetch()
+        } catch {
+            DDLogError("⛔️ Unable to fetch ShippingLabel for Site \(siteID) and Order \(order.orderID): \(error)")
```
Comment (jaclync) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsResultsControllers.swift]: super nit:

```suggestion
            DDLogError("⛔️ Unable to fetch ShippingLabels for Site \(siteID) and Order \(order.orderID): \(error)")
```

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsResultsControllers.swift [hunk 5]
```diff
@@ -239,7 +251,11 @@ private extension OrderDetailsResultsControllers {
             onReload()
         }
 
-        try? refundResultsController.performFetch()
+        do {
+            try refundResultsController.performFetch()
+        } catch {
+            DDLogError("⛔️ Unable to fetch Refund details for Site \(siteID) and Order \(order.orderID): \(error)")
```
Comment (jaclync) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsResultsControllers.swift]: super nit:

```suggestion
            DDLogError("⛔️ Unable to fetch Refunds for Site \(siteID) and Order \(order.orderID): \(error)")
```

**Review by jaclync — APPROVED** [review]
Changes LGTM 👍 just some super nit suggestions on the log messages.

> **Question:** We also perform a call to [refetchAllResultsControllers](https://github.com/woocommerce/woocommerce-ios/blob/4e0c223fc32db61b33c6e7b1c5c28973cd025f2f/WooCommerce/Classes/ViewModels/Order%20Details/OrderDetailsResultsControllers.swift#L315-L326) in order to re-fetch all the results controllers if necessary, and this involves more than one results controller. I left the optional `try?` in this case as if we wrap that on a `do-catch` block, the first operation that returns nil would also halt the rest, but happy to get any feedback on a better way to deal with those.

To improve logging more thoroughly, we can also log the errors in `refetchAllResultsControllers`. To prevent code duplication, we can move each do-catch `*ResultsController.performFetch()` to a function and then call it from both the `configure*ResultsController` and `refetchAllResultsControllers`. WDYT?

**Comment by iamgabrielma** [comment]
Thanks for the review and catching the grammar errors! 😅 

> To improve logging more thoroughly, we can also log the errors in refetchAllResultsControllers. To prevent code duplication, we can move each do-catch *ResultsController.performFetch() to a function and then call it from both the configure*ResultsController and refetchAllResultsControllers. WDYT?

That sounds great! I'll merge these for the moment and leave the second bit for a different PR, as I'm experimenting with a helper function in [this issue](https://github.com/woocommerce/woocommerce-ios/issues/8823), as well as with [removing the optionality entirely here](https://github.com/woocommerce/woocommerce-ios/commits/experiment/remove-optionalality-for-sort-descriptors)

**Pull request merged**